postgres/rdspostgres/rdspostgres.go

```diff
@@ -53,6 +55,40 @@ type Params struct {
 	TraceOpts []ocsql.TraceOption
 }
 
+type URLOpener struct {
+	// TraceOpts contains options for OpenCensus.
+	TraceOpts []ocsql.TraceOption
+}
+
+// Scheme is the URL scheme rdspostgres registers its URLOpener under on
+// postgres.DefaultMux.
+const Scheme = "rdspostgres"
+
+func init() {
+	postgres.DefaultURLMux().RegisterPostgres(Scheme, &URLOpener{})
+}
+
+func (uo *URLOpener) OpenPostgresURL(ctx context.Context, u *url.URL) (*sql.DB, error) {
+	cf := new(rds.CertFetcher)
+	for k, _ := range u.Query() {
+		// Only permit parameters that do not conflict with other behavior.
+		if k == "user" || k == "password" || k == "dbname" || k == "host" || k == "port" || k == "sslmode" || k == "sslcert" || k == "sslkey" || k == "sslrootcert" {
+			return nil, fmt.Errorf("cloudpostgres: OpenPostgresURL: extra parameter %s not allowed; use Params fields instead", k)
+		}
+	}
+	vals := u.Query()
+	vals.Set("sslmode", "disable")
+	u.RawQuery = vals.Encode()
+
+	db := sql.OpenDB(connector{
```

Comment:
Instead of calling `sql.OpenDB`, this code should be wrapping the `Open` function in this package, which properly handles the parameters and such.

Reply:
So, i fixed most of the comments except this one, coz it doesn't seems right to me, we pars Url struct in OpenPostgresURL to turn it into Params struct just to make it Url again in Open function, or we shoulr rewright Open function or we shouldn't use it.

Reply:
I understand that is how it operates internally, but the MySQL driver gives us no other way to pass the data in directly other connection strings (see go-sql-driver/mysql#771 where I've tried to report a similar concern upstream). While there is a mild inefficiency here, we're trading that for simplicity of understanding: URL openers should always conceptually use the same constructor that the driver type uses. This allows us to gain test confidence in individual units rather than having to test multiple codepaths.

Reply:
Ok, i get it, pushed the final version. Thanks for explanation.